From 6bf6381c033a134287318d49e94e71ae3e8e198e Mon Sep 17 00:00:00 2001
From: Harry Mander <harry.mander@pg.canterbury.ac.nz>
Date: Mon, 17 Jun 2024 08:39:06 +1200
Subject: [PATCH] Fix infinite loop when plot ranges are very small

---
 implot.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/implot.cpp b/implot.cpp
index 597b043..8e5f67a 100644
--- a/implot.cpp
+++ b/implot.cpp
@@ -719,6 +719,8 @@ void Locator_Default(ImPlotTicker& ticker, const ImPlotRange& range, float pixel
     const double nice_range = NiceNum(range.Size() * 0.99, false);
     const double interval   = NiceNum(nice_range / (nMajor - 1), true);
     const double graphmin   = floor(range.Min / interval) * interval;
+    if (graphmin + interval == graphmin)
+        return;
     const double graphmax   = ceil(range.Max / interval) * interval;
     bool first_major_set    = false;
     int  first_major_idx    = 0;
-- 
2.34.1

